pandas/core/arrays/sparse/dtype.py

```diff
@@ -223,7 +223,7 @@ def construct_from_string(cls, string):
             raise TypeError(msg)
 
     @staticmethod
-    def _parse_subtype(dtype):
+    def _parse_subtype(dtype) -> Tuple[str, Union[str, bool]]:
```

Comment:
not necessarily for this PR, but the `Union[str, bool]` is kind of weird.  maybe split into separate return values with clearer types?

Reply:
can `dtype` be typed as `str`?

Reply:
maybe we could cast has_fill_value to bool before returning. cast as in the dynamic bool(...) sense not cast from typing?